### PR TITLE
Feature/tokens whitelist

### PR DESCRIPTION
**Pull request by xenide**
1. In this refactor I am also introducing this "new" state management tool called `unstated-next`, which I got to know and is used in `governance-frontend` 
2. It provides a cleaner, more elegant way to manage state for react/nextjs apps as opposed to using hooks and context 
3. For future refactors of `info`, `farm`, or even `swap`, we should aim to replace the existing hooks/context with `unstated` whenever possible/sensible

### CODE REVIEW

File: pages/index.js [hunk 1]
```diff
@@ -86,17 +90,20 @@ export default function Home() {
       setPairs(_pairs);
     };
     getPairs();
-  }, []);
+  }, [whiteListedTokens]);
 
   useEffect(() => {
+    if (!whiteListedTokens) return;
```
Comment (OliverNChalk) [inline pages/index.js]:
```suggestion
    if (!whiteListedTokens) { return; }
```

File: state/whitelist.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+            const result = await axios.get(FOUNDATION_WHITELIST_URL);
+            const whitelist = {};
+            result.data.forEach((token) => {
+                // Converts all addresses into checksummed addresses
+                const tokenContractAddress = getAddress(token.address);
+                whitelist[tokenContractAddress] = token;
+            })
+            // Need to include the WVET address manually as it is not in the foundation whitelist
+            whitelist[WVET[1].address] = WVET[1];
+            setWhitelistedTokens(whitelist);
```
Comment (OliverNChalk) [inline state/whitelist.js:24]: Is there a good reason to prefer an object over a map? Like for logging the object or something? In general, I find the map API far cleaner and more modern (although a lot of the benefits come in TypeScript specifically).

Comment (xenide) [reply state/whitelist.js:24]: You're right. It's better especially in typescript. So I changed the implementation from object to a `Map` instead

**Review by OliverNChalk — APPROVED** [review]

**Review by KennethAshley — APPROVED** [review]
Looks good! Nice use of a `Map` here.

**Pull request merged**